### ISSUE TITLE
Metric Names: Relative Information Gain vs. LogLoss Reduction

### ISSUE THREAD

**Issue by rogancarr** [dotnet/machinelearning#2689]
In our Binary Classification and Multiclass Classification metrics, we have a measure for Relative Information Gain called `LogLossReduction`. Relative information gain seems to be the more commonly used term &mdash; it's actually hard to find search results on log-loss reduction. I wanted to get a quick set of opinions on:
1) Is this actually common nomenclature?
2) If not, shall we rename this to be Relative Information Gain?

**Comment by rogancarr** [comment]
@TomFinley @yaeldekel @justinormont Your opinions would be welcome.

**Comment by TomFinley** [comment]
When I was a wee little PhD student the preferred term was definitely "log loss," which was presumably why someone (Misha?) preferred this term. But this was a decade ago. If you are saying that nowadays the cool kids prefer the term "relative information gain," I have no means or desire to contradict you, and I am fine with it. I might only ask, can you cite some other authority to back up this claim? Even without this authority I would not object -- relative information gain seems fine to me -- but I would feel more easy in my mind with some appeal to a "higher" authority.

**Comment by codemzs** [comment]
Relative Information Gain seems a little more informative ...

**Comment by rogancarr** [comment]
@TomFinley `LogLoss` is certainly a thing, and is different than `LogLossReduction`. Actually, the "cool kids" usually call it the "cross-entropy" now, as that's what most neural net packages call it, and they are basically the same thing.

Let me clarify the loss functions / metrics that I'm talking about:

The **log loss** for binary classification is this:
![image](https://user-images.githubusercontent.com/19598317/53257285-9976f500-367e-11e9-8207-840045717172.png)
And for multiclass is this:
![image](https://user-images.githubusercontent.com/19598317/53257348-bca1a480-367e-11e9-8308-af6b2307f134.png)

The **log loss reduction**, also known as "relative log-loss", "relative information gain", and "reduction in information gain", is the log loss normalized  by the entropy (aka the log loss for a predictor predicting the prior). The "relative log loss" is calculated so:
![image](https://user-images.githubusercontent.com/19598317/53258798-5f0f5700-3682-11e9-964d-40454d492f0b.png)
where the 
![image](https://user-images.githubusercontent.com/19598317/53258967-cd541980-3682-11e9-8f25-67845e8fab39.png)
, and all logs here are in base 2. The idea is that a log-loss reduction / relative log-loss of 0.2 means that you did 20% better than guessing the prior.

So with that said, the term log-loss reduction doesn't seem to be in wide use. Relative log-loss is much more widely used, so maybe that's a better term. Relative information gain is great, but it's definitely an information-theory term and less of an ML term.

So what about we call it Relative Log Loss?

**Comment by najeeb-kazmi** [comment]
Is there any agreement on whether to call this Relative Log Loss or Relative Information Gain?

**Comment by abgoswam** [comment]
Discussed with @rogancarr and closing this issue.

We feel it is best to use documentation to  explain this metric, rather than renaming it to `RelativeLogLoss`, which is not a standard metric name either.
